hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/query/QueryDispatcher.java

```diff
@@ -0,0 +1,227 @@
+    }
+
+    /**
+     * Returns the {@link QueryHandler} for a given {@link Query}
+     *
+     * @param query the {@link Query} for which the {@link QueryHandler} is requested
+     * @return the {@code QueryHandler} for the query
+     */
+    @NonNull
+    public QueryHandler getHandler(@NonNull final Query query) {
+        return switch (query.getQueryCase()) {
+            case CONSENSUSGETTOPICINFO -> handlers.consensusGetTopicInfoHandler();
+
+            case GETBYSOLIDITYID -> handlers.contractGetBySolidityIDHandler();
+            case CONTRACTCALLLOCAL -> handlers.contractCallLocalHandler();
+            case CONTRACTGETINFO -> handlers.contractGetInfoHandler();
+            case CONTRACTGETBYTECODE -> handlers.contractGetBytecodeHandler();
+            case CONTRACTGETRECORDS -> handlers.contractGetRecordsHandler();
+
+            case CRYPTOGETACCOUNTBALANCE -> handlers.cryptoGetAccountBalanceHandler();
+            case CRYPTOGETINFO -> handlers.cryptoGetAccountInfoHandler();
+            case CRYPTOGETACCOUNTRECORDS -> handlers.cryptoGetAccountRecordsHandler();
+            case CRYPTOGETLIVEHASH -> handlers.cryptoGetLiveHashHandler();
+            case CRYPTOGETPROXYSTAKERS -> handlers.cryptoGetStakersHandler();
+
+            case FILEGETCONTENTS -> handlers.fileGetContentsHandler();
+            case FILEGETINFO -> handlers.fileGetInfoHandler();
+
+            case ACCOUNTDETAILS -> handlers.networkGetAccountDetailsHandler();
+            case GETBYKEY -> handlers.networkGetByKeyHandler();
+            case NETWORKGETVERSIONINFO -> handlers.networkGetVersionInfoHandler();
+            case NETWORKGETEXECUTIONTIME -> handlers.networkGetExecutionTimeHandler();
+            case TRANSACTIONGETRECEIPT -> handlers.networkTransactionGetReceiptHandler();
+            case TRANSACTIONGETRECORD -> handlers.networkTransactionGetRecordHandler();
+
+            case SCHEDULEGETINFO -> handlers.scheduleGetInfoHandler();
+
+            case TOKENGETINFO -> handlers.tokenGetInfoHandler();
+            case TOKENGETACCOUNTNFTINFOS -> handlers.tokenGetAccountNftInfosHandler();
+            case TOKENGETNFTINFO -> handlers.tokenGetNftInfoHandler();
+            case TOKENGETNFTINFOS -> handlers.tokenGetNftInfosHandler();
+
+            case TRANSACTIONGETFASTRECORD -> throw new UnsupportedOperationException(
+                    GET_FAST_RECORD_IS_NOT_SUPPORTED);
+            case QUERY_NOT_SET -> throw new UnsupportedOperationException(QUERY_NOT_SET);
+        };
+    }
+
+    /**
+     * Dispatch a validate-request. It is forwarded to the correct handler, which takes care of the
+     * specific functionality
+     *
+     * @param state the {@link HederaState} of this request
+     * @param query the {@link Query} of the request
+     * @throws NullPointerException if one of the arguments is {@code null}
+     */
+    public void dispatchValidate(@NonNull final HederaState state, @NonNull final Query query)
```

Comment:
Similarly could rename to `dispatchForValidation` ? or just ` validate` ?

Reply:
Updated